### ISSUE TITLE
ovn-kubernetes ovnkube-master containers crashlooping after 4.11.0-0.okd-2022-10-15-073651 update

### ISSUE THREAD

**Issue by nate-duke** [okd-project/okd#1372]
**Describe the bug**

The ovn-kubernetes ovnkube-master containers are continuously crashlooping since we updated to 4.11.0-0.okd-2022-10-15-073651.

Bit of log:
```
] [] []  [{kubectl-client-side-apply Update networking.k8s.io/v1 2022-09-12 12:25:06 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}}},"f:spec":{"f:ingress":{},"f:policyTypes":{}}} }]},Spec:NetworkPolicySpec{PodSelector:{map[] []},Ingress:[]NetworkPolicyIngressRule{NetworkPolicyIngressRule{Ports:[]NetworkPolicyPort{},From:[]NetworkPolicyPeer{NetworkPolicyPeer{PodSelector:&v1.LabelSelector{MatchLabels:map[string]string{access: true,},MatchExpressions:[]LabelSelectorRequirement{},},NamespaceSelector:nil,IPBlock:nil,},},},},Egress:[]NetworkPolicyEgressRule{},PolicyTypes:[Ingress],},} &NetworkPolicy{ObjectMeta:{allow-from-openshift-ingress  compsci-gradcentral  a405f843-c250-40d7-8dd4-a759f764f091 217304038 1 2022-09-22 14:36:38 +0000 UTC <nil> <nil> map[] map[] [] []  [{openshift-apiserver Update networking.k8s.io/v1 2022-09-22 14:36:38 +0000 UTC FieldsV1 {"f:spec":{"f:ingress":{},"f:policyTypes":{}}} }]},Spec:NetworkPolicySpec{PodSelector:{map[] []},Ingress:[]NetworkPolicyIngressRule{NetworkPolicyIngressRule{Ports:[]NetworkPolicyPort{},From:[]NetworkPolicyPeer{NetworkPolicyPeer{PodSelector:nil,NamespaceSelector:&v1.LabelSelector{MatchLabels:map[string]string{policy-group.network.openshift.io/ingress: ,},MatchExpressions:[]LabelSelectorRequirement{},},IPBlock:nil,},},},},Egress:[]NetworkPolicyEgressRule{},PolicyTypes:[Ingress],},}]: cannot clean up egress default deny ACL name: error in transact with ops [{Op:mutate Table:Port_Group Row:map[] Rows:[] Columns:[] Mutations:[{Column:acls Mutator:delete Value:{GoSet:[{GoUUID:60cb946a-46e9-4623-9ba4-3cb35f018ed6}]}}] Timeout:<nil> Where:[where column _uuid == {ccdd01bf-3009-42fb-9672-e1df38190cd7}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:mutate Table:Port_Group Row:map[] Rows:[] Columns:[] Mutations:[{Column:acls Mutator:delete Value:{GoSet:[{GoUUID:60cb946a-46e9-4623-9ba4-3cb35f018ed6}]}}] Timeout:<nil> Where:[where column _uuid == {10bbf229-8c1b-4c62-b36e-4ba0097722db}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:delete Table:ACL Row:map[] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[where column _uuid == {7b55ba0c-150f-4a63-9601-cfde25f29408}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:delete Table:ACL Row:map[] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[where column _uuid == {60cb946a-46e9-4623-9ba4-3cb35f018ed6}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:}] results [{Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:0 Error:referential integrity violation Details:cannot delete ACL row 7b55ba0c-150f-4a63-9601-cfde25f29408 because of 1 remaining reference(s) UUID:{GoUUID:} Rows:[]}] and errors []: referential integrity violation: cannot delete ACL row 7b55ba0c-150f-4a63-9601-cfde25f29408 because of 1 remaining reference(s)
```

```bash
NAME                   READY   STATUS             RESTARTS          AGE
ovnkube-master-cdnhd   6/6     Running            874 (6m35s ago)   3d
ovnkube-master-cmkvh   5/6     CrashLoopBackOff   858 (30s ago)     3d
ovnkube-master-r8fmh   6/6     Running            839 (36m ago)     3d
```

**Version**

Server Version: 4.11.0-0.okd-2022-10-15-073651
Kubernetes Version: v1.24.0-2566+5157800f2a3bc3-dirty
IPI on vsphere

**How reproducible**

Unknown. The crashlooping is continuous though.

**Log bundle**

[must-gather](https://nbc9-snips.cloud.duke.edu/snips/must-gather.local.2859117512952590880.zip)

**Comment by nate-duke** [comment]
This continues in `4.11.0-0.okd-2022-10-28-153352`. Happy to provide an update must-gather if necessary.

**Comment by vrutkovs** [comment]
A lot of weird things in this must-gather:
* `kube-rbac-proxy` is unable to connect:
```
2022-10-21T12:13:48.899318091Z 2022/10/21 12:13:48 http: proxy error: dial tcp 127.0.0.1:29102: connect: connection refused
```
* nbdb error:
```
2022-10-24T12:31:07.930975593Z 2022-10-24T12:31:07.930Z|29148|stream_ssl|WARN|SSL_read: unexpected SSL connection close
2022-10-24T12:31:07.931039628Z 2022-10-24T12:31:07.931Z|29149|jsonrpc|WARN|ssl:10.138.5.244:36376: receive error: Protocol error
```
* Several `Duplicate IP set: 10.....` messages in northd
* Most important I think is

**Comment by nate-duke** [comment]
> Seems OVN is choking on this NetworkPolicy. Could you check if it revives if its removed?

That specific NetworkPolicy is just deployed as a set via the [default network policy mechanism ](https://docs.okd.io/4.11/networking/network_policy/default-network-policy.html#nw-networkpolicy-project-defaults_default-network-policy). I can delete that specific policy but I've seen others mentioned as the problematic ones in the crash logs before.

I'll see if anybody in your jira is interested in taking a look.

**Comment by nate-duke** [comment]
I deleted the specific mentioned networkpolicy and watched the ovnkube-master container crash a few minutes later citing a different policy all together. it seems to be relatively random:
```
] [] []  [{kubectl-client-side-apply Update networking.k8s.io/v1 2022-09-12 12:25:06 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}}},"f:spec":{"f:ingress":{},"f:policyTypes":{}}} }]},Spec:NetworkPolicySpec{PodSelector:{map[] []},Ingress:[]NetworkPolicyIngressRule{NetworkPolicyIngressRule{Ports:[]NetworkPolicyPort{},From:[]NetworkPolicyPeer{NetworkPolicyPeer{PodSelector:&v1.LabelSelector{MatchLabels:map[string]string{access: true,},MatchExpressions:[]LabelSelectorRequirement{},},NamespaceSelector:nil,IPBlock:nil,},},},},Egress:[]NetworkPolicyEgressRule{},PolicyTypes:[Ingress],},} &NetworkPolicy{ObjectMeta:{allow-same-namespace  compsci-redis  8ea45cc1-a6da-480b-80ef-8ce701acfc08 216083137 1 2022-09-21 15:53:45 +0000 UTC <nil> <nil> map[] map[] [] []  [{openshift-apiserver Update networking.k8s.io/v1 2022-09-21 15:53:45 +0000 UTC FieldsV1 {"f:spec":{"f:ingress":{},"f:policyTypes":{}}} }]},Spec:NetworkPolicySpec{PodSelector:{map[] []},Ingress:[]NetworkPolicyIngressRule{NetworkPolicyIngressRule{Ports:[]NetworkPolicyPort{},From:[]NetworkPolicyPeer{NetworkPolicyPeer{PodSelector:&v1.LabelSelector{MatchLabels:map[string]string{},MatchExpressions:[]LabelSelectorRequirement{},},NamespaceSelector:nil,IPBlock:nil,},},},},Egress:[]NetworkPolicyEgressRule{},PolicyTypes:[Ingress],},} &NetworkPolicy{ObjectMeta:{deny-by-default  dws-geo-courses-globaled-test  74ccddfa-30f6-4f36-985d-412213436fb0 278123293 1 2022-10-31 09:39:54 +0000 UTC <nil> <nil> map[] map[] [] []  [{openshift-apiserver Update networking.k8s.io/v1 2022-10-31 09:39:54 +0000 UTC FieldsV1 {"f:spec":{"f:ingress":{},"f:policyTypes":{}}} }]},Spec:NetworkPolicySpec{PodSelector:{map[] []},Ingress:[]NetworkPolicyIngressRule{NetworkPolicyIngressRule{Ports:[]NetworkPolicyPort{},From:[]NetworkPolicyPeer{NetworkPolicyPeer{PodSelector:&v1.LabelSelector{MatchLabels:map[string]string{access: true,},MatchExpressions:[]LabelSelectorRequirement{},},NamespaceSelector:nil,IPBlock:nil,},},},},Egress:[]NetworkPolicyEgressRule{},PolicyTypes:[Ingress],},}]: cannot clean up egress default deny ACL name: error in transact with ops [{Op:mutate Table:Port_Group Row:map[] Rows:[] Columns:[] Mutations:[{Column:acls Mutator:delete Value:{GoSet:[{GoUUID:60cb946a-46e9-4623-9ba4-3cb35f018ed6}]}}] Timeout:<nil> Where:[where column _uuid == {ccdd01bf-3009-42fb-9672-e1df38190cd7}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:mutate Table:Port_Group Row:map[] Rows:[] Columns:[] Mutations:[{Column:acls Mutator:delete Value:{GoSet:[{GoUUID:60cb946a-46e9-4623-9ba4-3cb35f018ed6}]}}] Timeout:<nil> Where:[where column _uuid == {10bbf229-8c1b-4c62-b36e-4ba0097722db}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:delete Table:ACL Row:map[] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[where column _uuid == {60cb946a-46e9-4623-9ba4-3cb35f018ed6}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:} {Op:delete Table:ACL Row:map[] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[where column _uuid == {7b55ba0c-150f-4a63-9601-cfde25f29408}] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:}] results [{Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:1 Error: Details: UUID:{GoUUID:} Rows:[]} {Count:0 Error:referential integrity violation Details:cannot delete ACL row 7b55ba0c-150f-4a63-9601-cfde25f29408 because of 1 remaining reference(s) UUID:{GoUUID:} Rows:[]}] and errors []: referential integrity violation: cannot delete ACL row 7b55ba0c-150f-4a63-9601-cfde25f29408 because of 1 remaining reference(s)
```

**Comment by marqsbla** [comment]
I'm facing the same issue since few releases of OKD. Too many networkpolicies crashes the network in the cluster. What is working for me is deleting networkpolicies and reseting the network using redhat instructions: 
https://access.redhat.com/articles/6963671

I found a some issues that might be related to this problem:
https://github.com/okd-project/okd/discussions/1110
https://bugzilla.redhat.com/show_bug.cgi?id=1917240
https://bugzilla.redhat.com/show_bug.cgi?id=2045577

The problems I had: 
- container cannot be created, 
- the interface cannot be attached 
- The pod lost connectivity from the world (but liveness proble still works). 
- The more network policies I have the bigger problems I have. 
- Also pods from namespaces without network polices can be affected. 

I deleted all networkpolicies from the cluster and the problems stopped to occur.

**Comment by nate-duke** [comment]
Interesting. Thanks for the details from your environment. It's curious to me that OKD cannot operate normally with a few network policies per namespace given that 'namespace network isolation' is sort-of a hallmark for Openshift generally. I saw the warnings about 'large numbers of networkpolicies' but didn't think that several dozen was a large number

Does OCP not have this problem?

**Comment by msteenhu** [comment]
OCP only seems to support Openshift SDN last time I installed a 4.11 cluster, last weekend. OVN-Kubernetes turns out to be a lot more 'experimental' at this stage but we are also stuck with it.

**Comment by marqsbla** [comment]
that is kind of strange. I usually checked the documentation on OKD and OpenShift. They seemed the same in all cases. But here in the installation section they are different. OKD has default OVN, whereas Openshift OpenshiftSDN as the network driver. That is shame that I didn't double check it... 

https://docs.openshift.com/container-platform/4.11/installing/installing_openstack/installing-openstack-installer-custom.html#installation-configuration-parameters-network_installing-openstack-installer-custom
https://docs.okd.io/latest/installing/installing_openstack/installing-openstack-installer-custom.html#installation-configuration-parameters-network_installing-openstack-installer-custom

**Comment by vrutkovs** [comment]
>OCP only seems to support Openshift SDN last time I installed a 4.11 cluster,

OCP supports both OpenshiftSDN and OVN, but OKD defaults to OVN. Soon they both they'd default to OVN.

**Comment by nate-duke** [comment]
relayed to https://issues.redhat.com/browse/OCPBUGS-3245

**Comment by marqsbla** [comment]
did you try running this: https://access.redhat.com/articles/6963671 ?

**Comment by nate-duke** [comment]
> did you try running this: https://access.redhat.com/articles/6963671 ?

```
WARNING: This procedure will cause cluster-wide network interruption and has some risk. Please perform it only if instructed to do so or in case of OVN-Kubernetes issues.
```

No, I did not.

**Comment by vrutkovs** [comment]
OVN bug was fixed in 4.11+, should be available in latest 4.12 stable
